### ISSUE TITLE
CentOS 8 rpm build fails on documentation directory

### ISSUE THREAD

**Issue by dsavineau** [nfs-ganesha/nfs-ganesha#631]
It looks like since [1] (not quite sure but this is the only related commit I found even if this was something specific to SLES and not CentOS) the documentation directories aren't created anymore resulting a failure in the rpm build on CentOS 8.

```console
DEBUG: RPM build errors:
DEBUG: error: Directory not found: /builddir/build/BUILDROOT/nfs-ganesha-4-dev.31.el8.x86_64/usr/share/doc/ganesha
DEBUG: error: File not found: /builddir/build/BUILDROOT/nfs-ganesha-4-dev.31.el8.x86_64/usr/share/doc/ganesha/*
DEBUG:     Directory not found: /builddir/build/BUILDROOT/nfs-ganesha-4-dev.31.el8.x86_64/usr/share/doc/ganesha
DEBUG:     File not found: /builddir/build/BUILDROOT/nfs-ganesha-4-dev.31.el8.x86_64/usr/share/doc/ganesha/*
DEBUG: Child return code was: 1
```

The `/usr/share/doc/ganesha` directory is still referenced in the spec file [2]

```console
# rpm --eval "%{_defaultdocdir}"
/usr/share/doc
```

This is on the next branch (V4-dev.31)

You can found the full build log at [3]

Note that the V4-dev.xx was working fine few weeks ago. I can't find the precise moment when this was starting to fail and could match [1] but at least one successfull job can be found at [4] (V4-dev.24)

[1] https://github.com/nfs-ganesha/nfs-ganesha/commit/0eb5c4a1668e5a2b2a76793c14131e15077effa1
[2] https://github.com/nfs-ganesha/nfs-ganesha/blob/next/src/nfs-ganesha.spec-in.cmake#L671-L672
[3] https://jenkins.ceph.com/view/all/job/nfs-ganesha/1479/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos8,DIST=centos8,MACHINE_SIZE=huge/consoleFull
[4] https://jenkins.ceph.com/view/all/job/nfs-ganesha/lastSuccessfulBuild/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos8,DIST=centos8,MACHINE_SIZE=huge/consoleFull

**Comment by dsavineau** [comment]
Just found the latest CentOS 8 successfull job [1] and was using V4-dev.29

The next job [2] on august 13 failed. It was based on V4-dev.30 which includes the commit mentioned earlier.

[1] https://jenkins.ceph.com/view/all/job/nfs-ganesha/1437/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos8,DIST=centos8,MACHINE_SIZE=huge/consoleFull
[2] https://jenkins.ceph.com/view/all/job/nfs-ganesha/1438/

**Comment by dsavineau** [comment]
@supriti @ffilz

**Comment by supriti** [comment]
@dsavineau I removed it only for SLES.  Spec file was expecting files to be installed at "/usr/share/doc/packages/ganesha" but instead it was installed at "/usr/share/doc/ganesha". Also, the logic was that when we install any nfs-ganesha FSAL package, the respective config file is installed at /etc/ganesha/. So there is no need to install all the sample config files. In downstream spec file earlier we had all the config files listed under section %files (as rpm complains if there are missing files are that installed but not packaged). I removed that part from downstream spec file. 

Spec file looks like this for us:
```
%install
mkdir -p %{buildroot}%{_docdir}/ganesha
install -m 644 LICENSE.txt %{buildroot}%{_docdir}/ganesha/LICENSE.txt

%files
%dir %{_docdir}/ganesha
%{_docdir}/ganesha/LICENSE.txt
```

I don't know why it breaks for CentOS though. Maybe we need to update spec file as well.
